rviz_rendering/src/rviz_rendering/ogre_render_window_impl.cpp

```diff
@@ -260,13 +260,8 @@ RenderWindowImpl::resize(size_t width, size_t height)
 {
   if (ogre_render_window_) {
     this->setCameraAspectRatio();
-    ogre_render_window_->resize(
-      static_cast<unsigned int>(width),  // NOLINT
-      static_cast<unsigned int>(height)  // NOLINT
-    );
     ogre_render_window_->windowMovedOrResized();
   }
-  this->renderLater();
```

Comment:
Could you elaborate on why this change? Sorry, it's not clear to me exactly what's going on.

Reply:
the resize logic before was duplicated in multiple different places. this PR cleans it up to one function so should be easier to maintain / understand.

Reply:
honestly working with this qt logic was very hard to understand also 😄

Reply:
To confirm, are you saying we do not need to call `Ogre::RenderWindow::resize` at all? I don't see another place in the code where this is happening.

Reply:
I think that's right. Yes. I'm not a QtWindow expert though.

Reply:
no. what I'm saying is the resize logic before had multiple different places where it did the same thing. I am just trying to clean it up.

Reply:
Sounds good, however, previously we were calling `Ogre::RenderWindow::resize` (on `ogre_render_window_`), but now I don't see any place in the code that calls that API. Can you point me to where the resize is happening or maybe this particular API call is not necessary? AFAICT, it looks like we're no longer calling `resize` on the underlying Ogre render window.

Reply:
> AFAICT, it looks like we're no longer calling resize on the underlying Ogre render window.

yes I think this is right. we dont have to call the `resize()` function at all. looking at the [Qt5 example for the QWindow](https://doc.qt.io/qt-5/qtgui-rasterwindow-example.html) I dont think it is necessary.....as long as you have implemented the `resizeEvent` callback. Before `rviz` and `rviz2` had not implemented the `resizeEvent` callback because *I think* it was handled within the `event()` callback using a switch case.

this PR was meant to switch the logic over to something more similar to what the example shows.

Reply:
Sorry, I'm still not convinced :slightly_smiling_face: I think we're talking past each other. The resize call removed at L263 (removed) above is a pointer to an [`Ogre::RenderWindow`](https://ogrecave.github.io/ogre/api/1.12/class_ogre_1_1_render_window.html), which is different than a QWindow; it doesn't have a `resizeEvent` method.

I'll try playing around with some rendered objects and resizing to confirm that this PR is indeed not changing behavior.

Reply:
>I'll try playing around with some rendered objects and resizing to confirm that this PR is indeed not changing behavior.

I actually observe a slight improvement while resizing the window on my machine. not sure if you'll see the same but you might. it seems more "snappy" i guess.